lib/utils/__tests__/report.test.mjs

```diff
@@ -550,3 +563,304 @@ test('without node nor line', () => {
+				type: 'DeprecationWarning',
+			}),
+		);
+	});
+
+	it('has a non-zero endIndex', () => {
+		const v = {
+			ruleName: 'foo',
+			result: {
+				warn: jest.fn(),
+				stylelint: resultStylelint(),
+			},
+			message: 'bar',
+			node: {
+				rangeBy: defaultRangeBy,
+			},
+			endIndex: 1,
+		};
+
+		report(v);
+
+		expect(emitWarning).toHaveBeenCalledWith(
+			expect.stringMatching(
+				/Partial position information in the `utils.report\(\)` function is deprecated \("foo"\)\./,
+			),
+			expect.objectContaining({
+				code: 'stylelint:007',
+				detail:
+					'Please pass both `index` and `endIndex` as arguments in the `utils.report()` function of "foo".',
+				type: 'DeprecationWarning',
+			}),
+		);
+	});
+});
+
+describe('with incorrect start or end', () => {
+	it('has no start', () => {
+		const v = {
+			ruleName: 'foo',
+			result: {
+				warn: jest.fn(),
+				stylelint: resultStylelint(),
+			},
+			message: 'bar',
+			node: {
+				rangeBy: defaultRangeBy,
+			},
+			end: {
+				line: 1,
+				column: 2,
+			},
+		};
+
+		report(v);
+
+		expect(emitWarning).toHaveBeenCalledWith(
+			expect.stringMatching(
+				/Partial position information in the `utils.report\(\)` function is deprecated \("foo"\)\./,
+			),
+			expect.objectContaining({
+				code: 'stylelint:007',
+				detail:
+					'Please pass both a valid `start` and `end` argument in the `utils.report()` function of "foo".',
+				type: 'DeprecationWarning',
+			}),
+		);
+	});
+
+	it('has an invalid start', () => {
+		const v = {
+			ruleName: 'foo',
+			result: {
+				warn: jest.fn(),
+				stylelint: resultStylelint(),
+			},
+			message: 'bar',
+			node: {
+				rangeBy: defaultRangeBy,
+			},
+			start: {
+				line: 1,
+			},
+			end: {
+				line: 1,
+				column: 2,
+			},
+		};
+
+		report(v);
+
+		expect(emitWarning).toHaveBeenCalledWith(
+			expect.stringMatching(
+				/Partial position information in the `utils.report\(\)` function is deprecated \("foo"\)\./,
+			),
+			expect.objectContaining({
+				code: 'stylelint:007',
+				detail:
+					'Please pass both a valid `start` and `end` argument in the `utils.report()` function of "foo".',
+				type: 'DeprecationWarning',
+			}),
+		);
+	});
+
+	it('has no end', () => {
+		const v = {
+			ruleName: 'foo',
+			result: {
+				warn: jest.fn(),
+				stylelint: resultStylelint(),
+			},
+			message: 'bar',
+			node: {
+				rangeBy: defaultRangeBy,
+			},
+			start: {
+				line: 1,
+				column: 1,
+			},
+		};
+
+		report(v);
+
+		expect(emitWarning).toHaveBeenCalledWith(
+			expect.stringMatching(
+				/Partial position information in the `utils.report\(\)` function is deprecated \("foo"\)\./,
+			),
+			expect.objectContaining({
+				code: 'stylelint:007',
+				detail:
+					'Please pass both a valid `start` and `end` argument in the `utils.report()` function of "foo".',
+				type: 'DeprecationWarning',
+			}),
+		);
+	});
+
+	it('has an invalid end', () => {
```

Comment:
question

It seems that you have covered many cases, is it our responsibility if `start` comes after `end`?
e.g.
```
start: {
	line: 5,
	column: 3,
},
end: {
	line: 1,
	column: 2,
},
```

Reply:
I think that is out of scope :)

We could investigate how PostCSS handles such arguments, both with `start/end` and `index/endIndex`. Can you open an issue about this?

Reply:
On `postcss` repository?

Reply:
I would first open an issue here to investigate. Then we don't block this PR as this question is out of scope. If we do find that improvements can be made in such cases then we can see where and how to submit patches.

On the other hand I think it is a fair assumption that `start` always comes before `end`. If rule authors get it wrong then they will see this in their test outcomes. We already have systems in place to validate the exact ranges.

So honestly I think this is a non-issue :)